src/node/status_node.py

```diff
@@ -76,6 +76,23 @@ def get_pubkey(self):
     def wait_fully_started(self):
         assert self.search_logs(string="retrieve messages...")
 
+    def wait_for_logs(self, strings=None, timeout=10):
+        if strings is not None and not isinstance(strings, list):
+            raise ValueError("strings must be a list")
+        start_time = time.time()
+        while time.time() - start_time < timeout:
+            all_found = True
+            if strings:
```

Comment:
Given we already asserted `strings` is a list, this check is not needed. If `strings` is an empty list, the loop will skip it.